pkg/test/amd_tests.go

```diff
@@ -0,0 +1,923 @@
+		&testSEVSNPSideChannelProtection,
+		&testSEVSNPFirmwareVersion,
+		&testSEVSNPVMBoot,
+		&testSEVSNPAttestation,
+	}
+
+	TestsAMDPSP = []*Test{
+		&testPSBStatus,
+		&testPSBEnabled,
+		&testPlatformVendorID,
+		&testPlatformModelID,
+		&testBIOSKeyRevision,
+		&testAMDKeyDisabled,
+		&testSecureDebugDisabled,
+		&testKeysFused,
+		&testPSBPolicyHash,
+		&testRevocationStatus,
+	}
+
+	TestsAMDSME = []*Test{
+		&testSMESupport,
+		&testSMEEnabled,
+		&testSMEKernelOption,
+		&testSMEKernelCommandline,
+		&testSMEFunctionality,
+		&testTSMESupport,
+		&testTSMEEnabled,
+	}
+
+	TestsAMDSEV = []*Test{
+		&testSEVSupport,
+		&testSEVEnabled,
+		&testSEVESupport,
+		&testSEVESEnabled,
+		&testSEVFirmwareVersion,
+		&testSEVGuestConfig,
+	}
+	TestsAMDSEVSNP = []*Test{
+		&testSEVSNPSupport,
+		&testSEVSNPEnabled,
+		&testSEVESupport,
+		&testSEVESEnabled,
+		&testSEVSNPDebugRegisters,
+		&testSEVSNPSideChannelProtection,
+		&testSEVSNPFirmwareVersion,
+		&testSEVSNPVMBoot,
+		&testSEVSNPAttestation,
+	}
+	TestsAMDGeneral = []*Test{
+		&testAMDFamilyModel,
+	}
+)
+
+// SME Tests
+
+// SMESupport checks if SME is supported
+func SMESupport(hw hwapi.LowLevelHardwareInterfaces, p *PreSet) (bool, error, error) {
+	eax, _, _, _ := hw.CPUID(CPUID_SME, 0)
+	if eax&CPUID_SME_SUPPORT == 0 {
+		return false, fmt.Errorf("SME is not supported on this CPU (CPUID.0x8000001F[EAX].bit[0] = 0)"), nil
+	}
+	log.Debugf("SME is supported (CPUID.0x8000001F[EAX].bit[0] = 1)")
+	return true, nil, nil
+}
+
+// SMEEnabled checks if SME is enabled
+func SMEEnabled(hw hwapi.LowLevelHardwareInterfaces, p *PreSet) (bool, error, error) {
+	vals := hw.ReadMSR(MSR_AMD64_SYSCFG)
+	if len(vals) == 0 {
+		return false, fmt.Errorf("ReadMSR returned no values"), nil
+	}
+	val := vals[0]
+	if val&MSR_AMD64_SYSCFG_SME_EN == 0 {
+		return false, fmt.Errorf("SME is not enabled (MSR 0xC0010010[23] = 0)"), nil
+	}
+	log.Debugf("SME is enabled (MSR 0xC0010010[23] = 1)")
+	return true, nil, nil
+}
+
+// SMEKernelOption checks if SME kernel option is set
+func SMEKernelOption(hw hwapi.LowLevelHardwareInterfaces, p *PreSet) (bool, error, error) {
+	data, err := os.ReadFile("/proc/cmdline")
+	if err != nil {
+		return false, fmt.Errorf("failed to read /proc/cmdline: %v", err), nil
+	}
+	cmdline := string(data)
+	if !containsOption(cmdline, "mem_encrypt=on") {
+		return false, fmt.Errorf("SME kernel option 'mem_encrypt=on' is not set in kernel command line"), nil
+	}
+	log.Debugf("SME kernel option 'mem_encrypt=on' is set in kernel command line")
+	return true, nil, nil
+}
+
+// containsOption checks if a given option is present in the kernel command line
+func containsOption(cmdline, option string) bool {
+	for _, part := range splitCommandLine(cmdline) {
+		if part == option {
+			return true
+		}
+	}
+	return false
+}
+
+// splitCommandLine splits the kernel command line into options
+func splitCommandLine(cmdline string) []string {
+	return strings.Fields(cmdline)
+}
```

Comment:
This can be replaced inline with `strings.Contains(cmdline, option)` and these functions can be removed.